lib/bozo/compilers/msbuild.rb

```diff
@@ -184,10 +188,36 @@ def framework_version
       framework_version
     end
 
+    def windows_path
+      to_unix_path(ENV['WINDIR'])
+    end
+
+    def default_msbuild_path(config)
+      File.join(windows_path, 'Microsoft.NET', config[:framework], config[:version], 'msbuild.exe')
+    end
+
+    def program_files
+      globalArch = ENV['PROCESSOR_ARCHITECTURE']
+      to_unix_path(
+        ENV[globalArch == 'AMD64' ? 'ProgramFiles(x86)' : 'ProgramFiles'])
+    end
+
+    def msbuild_path
+      [ File.join(program_files, 'MSBuild/14.0/Bin'),
+        File.join(program_files, 'MSBuild/12.0/Bin'),
+        File.join(program_files, 'MSBuild/12.0/Bin/amd64'),
+        File.join(windows_path, 'Microsoft.NET/Framework/v4.0.30319'),
+        File.join(windows_path, 'Microsoft.NET/Framework/v4.0.30128'),
+        File.join(windows_path, 'Microsoft.NET/Framework/v3.5')].
+          collect {|p| File.join(p, 'MSBuild.exe')}.
+          find {|p| File.exists?(p)}
```

Comment:
It's possible to override the path in `bozorc.rb` using the new `msbuild_path` property.  I'm actually proposing to change that to a search path override.  Stay tuned.